plugins/inputs/ipmi_sensor/ipmi.go

```diff
@@ -143,13 +159,76 @@ func (m *Ipmi) parse(acc telegraf.Accumulator, server string) error {
 	return nil
 }
 
-func Atofloat(val string) float64 {
+func parseV2(acc telegraf.Accumulator, hostname string, cmdOut string) error {
+	// each line will look something like
+	// CMOS Battery     | 65h | ok  |  7.1 |
+	// Temp             | 0Eh | ok  |  3.1 | 55 degrees C
+	// Drive 0          | A0h | ok  |  7.1 | Drive Present
+	lines := strings.Split(cmdOut, "\n")
+	for i := 0; i < len(lines); i++ {
+		ipmiFields := extractFieldsFromRegex(`^(?P<name>[^|]*)\|[^|]+\|(?P<status_code>[^|]*)\|(?P<entity_id>[^|]*)\|(?:(?P<description>[^|]+))?`, lines[i])
+		if len(ipmiFields) < 3 || len(ipmiFields) > 4 {
+			continue
+		}
+
+		tags := map[string]string{
+			"name": transform(ipmiFields["name"]),
+		}
+
+		// tag the server is we have one
+		if hostname != "" {
+			tags["server"] = hostname
+		}
+		tags["entity_id"] = transform(ipmiFields["entity_id"])
+		tags["status_code"] = trim(ipmiFields["status_code"])
+		fields := make(map[string]interface{})
+		descriptionResults := extractFieldsFromRegex(`^(?P<analogValue>[0-9.]+)\s(?P<analogUnit>.*)|(?P<status>.+)|^$`, trim(ipmiFields["description"]))
+		// This is an analog value with a unit
+		if descriptionResults["analogValue"] != "" && len(descriptionResults["analogUnit"]) >= 1 {
+			fields["value"] = aToFloat(descriptionResults["analogValue"])
```

Comment:
I think if the value cannot be converted to a float then we should skip the line.